src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateManager.cs

```diff
@@ -100,64 +95,97 @@ private static X509Certificate2 CertificateFromThumbprint(X509Store store, strin
+        // Install the certificate into the My store.
         // It will not install the certificate if it is already present in the store.
         // It returns the thumbprint of the certificate, regardless whether it was added or found.
-        public static string InstallMyCertificate(BridgeConfiguration configuration, string certificateName)
+        public static string InstallCertificateToMyStore(X509Certificate2 certificate, X509Certificate2 rootCertificate)
         {
             // Installing any MY certificate guarantees the certificate authority is loaded first
-            InstallRootCertificate(configuration, configuration.BridgeCertificateAuthority);
+            InstallCertificateToRootStore(rootCertificate);
 
             lock (s_certificateLock)
             {
                 CertificateCacheEntry entry = null;
-                if (s_myCertificates.TryGetValue(certificateName, out entry))
+                if (s_myCertificates.TryGetValue(certificate.Subject, out entry))
                 {
-                    return entry.ThumbPrint;
+                    return entry.Thumbprint;
                 }
 
-                string certificateFilePath = CreateCertificateFilePath(configuration, certificateName);
-                X509Certificate2 cert = new X509Certificate2();
-                // "test" is currently the required password to allow exportable private keys
-                cert.Import(certificateFilePath, "test", X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet);
-
-                bool added = AddToStoreIfNeeded(StoreName.My, StoreLocation.LocalMachine, cert);
-                s_myCertificates[certificateName] = new CertificateCacheEntry
+                bool added = AddToStoreIfNeeded(StoreName.My, StoreLocation.LocalMachine, certificate);
+                s_myCertificates[certificate.Subject] = new CertificateCacheEntry
                 {
-                    ThumbPrint = cert.Thumbprint,
+                    Thumbprint = certificate.Thumbprint,
                     AddedToStore = added
                 };
 
-                return cert.Thumbprint;
+                return certificate.Thumbprint;
             }
         }
 
+        // When called, generates the a cert for the machine DNS name with SAN localhost, and installs both certs
+        // returns thumbprint of the machine certs
+        public static X509Certificate2 CreateAndInstallLocalMachineCertificates(CertificateGenerator certificateGenerator)
+        {
+            if (certificateGenerator == null)
+            {
+                throw new ArgumentNullException("certificateGenerator");
+            }
+
+            lock (s_certificateLock)
+            {
+                if (s_localCertificate != null)
+                {
+                    return s_localCertificate;
+                }
+
+                Trace.WriteLine("[CertificateManager] Installing Root and Machine certificates to machine store.");
+
```

Comment:
Nit: could use the more common trace pattern that includes a time stamp and category.